legacy_promotions/db/migrate/20161017102621_create_spree_promotion_code_batch.rb

```diff
@@ -2,6 +2,8 @@
 
 class CreateSpreePromotionCodeBatch < ActiveRecord::Migration[5.0]
   def change
+    return if table_exists?(:spree_promotion_code_batches)
+
     create_table :spree_promotion_code_batches do |t|
```

Comment:
I would prefer `if_not_exists: true` checks on the `create_table` calls and foreign_keys, etc.